k9mail/src/main/java/com/fsck/k9/controller/MessagingController.java

```diff
@@ -1622,11 +1622,12 @@ private <MSG extends Message> void updateStatus(
             final int todo,
             final LocalMessage localMessage,
             final MSG originalMessage,
+            final String messageId,
```

Comment:
this new parameter allows us then to decouple the message you pass from the one you remove from OngoingDecrypt as you can pass the shouldRemoveId flag to true and a different ID than the original message. Clear.

Reply:
> this new parameter allows us then to decouple the message you pass from the one you remove from OngoingDecrypt as you can pass the shouldRemoveId flag to true and a different ID than the original message. Clear.

Yes, the thing is that the "originalMessage" could even be null in the worst cases. Then we can ensure we will remove the messageId from the "decryption ongoing messages" list in the storage.
(If we don't remove this messageId, the app may be thinking that there was a crash during a message decryption and the app came back to life. This is reproducing specially with the MultipleX messages replied from the bot, because all of them come with same messageId!)